### PR TITLE
bug fix undefined index in stats section

### PR DESCRIPTION
**Pull request by pratik**
Signed-off-by: Pratik Sethia <pratiksethia1@gmail.com>

### CODE REVIEW

**Comment by pratiksethia1** [comment]
screenshot

File: app/Controller/StatsController.php [hunk 1]
```diff
@@ -66,8 +66,8 @@ public function stats() {
 	}
 
 	protected function _getTimeFilter() {
-		$filter = $this->Incident->filterTimes[$this->request->query('filter')];
 		if (isset($filter)) {
+			$filter = $this->Incident->filterTimes[$this->request->query('filter')];
```
Comment (nijel) [inline app/Controller/StatsController.php]: This really looks wrong to set the variable only after checking it is set...

File: app/Controller/StatsController.php [hunk 2]
```diff
@@ -66,9 +66,12 @@ public function stats() {
...
 	protected function _getTimeFilter() {
-		$filter = $this->Incident->filterTimes[$this->request->query('filter')];
-		if (isset($filter)) {
-			return $filter;
+		$getFilter=$this->request->query('filter');
+		if(isset($getFilter)) {
+			$filter = $this->Incident->filterTimes[$getFilter];
+			if (isset($filter)) {
+				return $filter;
+			}
```
Comment (nijel) [inline app/Controller/StatsController.php]: It should fallback to returing all as well here, otherwise the method does not return anything here...

**Comment by nijel** [comment]
Obsolete with upgrade to CakePHP 3.0.

**Pull request closed without merging**